src/Sdk/WebApi/WebApi/ResultsHttpClient.cs

```diff
@@ -214,9 +214,16 @@ private async Task UploadBlockFileAsync(string url, string blobStorageType, File
             if (m_useSdk && blobStorageType == BlobStorageTypes.AzureBlobStorage)
             {
                 var blobClient = GetBlobClient(url);
+                var uploadOptions = new BlobUploadOptions
+                {
+                    HttpHeaders = new BlobHttpHeaders
+                    {
+                        ContentType = "text/plain"
+                    }
+                };
                 try
                 {
-                    await blobClient.UploadAsync(file, cancellationToken);
+                    await blobClient.UploadAsync(file, uploadOptions, cancellationToken);
```

Comment:
do you need to check what is the default option is and merge with it instead of just overwrite it?

Reply:
not quite sure i follow your new change.

my initial comment is basically try to make sure the SDK don't have any special initialized `BlobUploadOptions`.

Based on the source code, maybe this is what you want?

Not sure... we can chat more...
```
await blobClient.UploadAsync(file, new BlobUploadOptions()
                    {
                        HttpHeaders = new BlobHttpHeaders
                        {
                            ContentType = "text/plain"
                        },
                        Conditions = new BlobRequestConditions
                        {
                            IfNoneMatch = new ETag("*")
                        }
                    }, cancellationToken);
```

Reply:
Gotcha! That makes sense to me. I was thinking specifically around the HTTP headers but I do recall seeing that in the source code of the SDK when I looked.

Reply:
Found this issue which was helpful for wrapping my head around the options: https://github.com/Azure/azure-sdk-for-net/issues/9823. Thanks for calling it out!